src/centos/scripts/start-marklogic.sh

```diff
@@ -283,39 +319,52 @@ fi
 ################################################################
 if [[ -f /var/opt/MarkLogic/DOCKER_JOIN_CLUSTER ]]; then
     info "MARKLOGIC_JOIN_CLUSTER is true, but skipping join because this instance has already joined a cluster."
-elif [[ "${MARKLOGIC_JOIN_CLUSTER}" == "true" ]] && [[ "${HOST_FQDN}" != "${MARKLOGIC_BOOTSTRAP_HOST}" ]]; then
-    info "MARKLOGIC_JOIN_CLUSTER is true and join conditions are met, joining host to the cluster."
-    
-    if [[ -z "${MARKLOGIC_GROUP}" ]]; then
-        info "MARKLOGIC_GROUP is not specified, adding host to the Default group."
-        MARKLOGIC_GROUP_PAYLOAD=\"group=Default\"
-    else
-        GROUP_RESP_CODE=$(curl --anyauth --user "${ML_ADMIN_USERNAME}":"${ML_ADMIN_PASSWORD}" -m 30 -s -o /dev/null -w "%{http_code}" -X GET http://"${MARKLOGIC_BOOTSTRAP_HOST}":8002/manage/v2/groups/"${MARKLOGIC_GROUP}")
-        if [[ ${GROUP_RESP_CODE} -eq 200 ]]; then
-            info "MARKLOGIC_GROUP is specified, adding host to the ${MARKLOGIC_GROUP} group."
-            MARKLOGIC_GROUP_PAYLOAD=\"group=${MARKLOGIC_GROUP}\"
+elif [[ "${MARKLOGIC_JOIN_CLUSTER}" == "true" ]]; then
+     # Validate bootsrap host before joining cluster
+    CHECK_BOOTSTRAP=$(validate_bootstrap "${MARKLOGIC_BOOTSTRAP_HOST}")  
+
+    if [[ "${CHECK_BOOTSTRAP}" == "valid" ]]; then
```

Comment:
As a general rule, to reduce the level of nested if statements and increase code readability, it is better to use one if statement without an else block to check if the condition is invalid.

Reply:
Using an if-else block is ok as per the best practice. I agree that the if nested-if blocks should be reduced for readability. The script is already using a lot of if-else blocks which should be refactored as well if possible. A spike for refactoring would be good.